server/src/main/java/io/crate/expression/RegexpFlagsHelper.java

```diff
@@ -1,89 +1,31 @@
+ * not use this file except in compliance with the License.  You may
  * obtain a copy of the License at
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
- * License for the specific language governing permissions and limitations
- * under the License.
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
  *
  * However, if you have executed another commercial license agreement
  * with Crate these terms will supersede the license and you may use the
- * software solely pursuant to the terms of the relevant commercial agreement.
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
  */
 
-package io.crate.expression.scalar.regex;
+package io.crate.expression;
 
 import javax.annotation.Nullable;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class RegexMatcher {
-
-    private final boolean globalFlag;
-    private final Pattern pattern;
-    private Matcher matcher;
-
-    public RegexMatcher(String regex, int flags, boolean globalFlag) {
-        pattern = Pattern.compile(regex, flags);
-        this.globalFlag = globalFlag;
-    }
-
-    public RegexMatcher(String regex, @Nullable String flags) {
-        this(regex, parseFlags(flags), isGlobal(flags));
-    }
-
-    public RegexMatcher(String regex) {
-        this(regex, 0, false);
-    }
-
-    public boolean match(String term) {
-        initMatcher(term);
-        return matcher.find();
-    }
-
-    @Nullable
-    public List<String> groups() {
-        int groupCount = matcher.groupCount();
-        if (groupCount == 0) {
-            try {
-                return List.of(matcher.group());
-            } catch (IllegalStateException e) {
-                return null;
-            }
-        }
-        ArrayList<String> groups = new ArrayList<>(groupCount);
-        for (int i = 0; i < groupCount; i++) {
-            groups.add(matcher.group(i + 1));
-        }
-        return groups;
-    }
-
-    public String replace(String term, String replacement) {
-        initMatcher(term);
-        if (globalFlag) {
-            return matcher.replaceAll(replacement);
-        } else {
-            return matcher.replaceFirst(replacement);
-        }
-    }
-
-    private void initMatcher(String term) {
-        if (matcher == null) {
-            matcher = pattern.matcher(term);
-        } else {
-            matcher.reset(term);
-        }
-    }
+public final class RegexpFlagsHelper {
```

Comment:
```suggestion
public final class RegexpFlags {
```

Not sure if `Helper` adds anything useful to readers.